BuildingBlocks/src/DevelopmentKit.Identity/ValueObjects/IdentityAddress.cs

```diff
@@ -45,37 +47,41 @@ public static bool IsValid(string? stringValue)
 
         var lengthIsValid = stringValue.Length <= MAX_LENGTH;
 
-        var realm = stringValue[..3];
-
-        var concatenation = Base58.Bitcoin.Decode(stringValue.AsSpan(3)).ToArray();
-        var hashedPublicKey = concatenation[..20];
-        var givenChecksum = concatenation[20..];
+        const string pattern = @"^did\:e\:(.+?)\:dids\:(.+)(.{2})$";
+        try
+        {
+            var matches = Regex.Matches(stringValue, pattern, RegexOptions.IgnoreCase).First().Groups;
```

Comment:
Hm, if there is no match, then `First` will throw an exception. So you should make sure that there is a match before getting it.

Reply:
This is still not fixed...